src/main/java/com/dqu/simplerauth/managers/CacheManager.java

```diff
@@ -0,0 +1,123 @@
+    private static final File DBFILE = new File(PATH);
+    private static JsonObject db = new JsonObject();
+
+    public static void loadCache() {
+        if (!DBFILE.exists()) {
+            db.addProperty("version", VERSION);
+
+            db.add("minecraft-account-cache", new JsonArray());
+
+            saveCache();
+        }
+
+        try {
+            BufferedReader bufferedReader = Files.newReader(DBFILE, StandardCharsets.UTF_8);
+            db = GSON.fromJson(bufferedReader, JsonObject.class);
+        } catch (Exception e) {
+            AuthMod.LOGGER.error(e);
+        }
+    }
+
+    private static void saveCache() {
+        try {
+            BufferedWriter bufferedWriter = Files.newWriter(DBFILE, StandardCharsets.UTF_8);
+            String json = GSON.toJson(db);
+            bufferedWriter.write(json);
+            bufferedWriter.close();
+        } catch (Exception e) {
+            AuthMod.LOGGER.error(e);
+        }
+    }
+
+    public static JsonArray getMinecraftAccounts() {
+        return db.getAsJsonArray("minecraft-account-cache");
+    }
+
+    @Nullable
+    public static JsonObject getMinecraftAccount(String username) {
+        removeExpired();
+        JsonArray minecraftAccounts = getMinecraftAccounts();
+        for (int i = 0; i < minecraftAccounts.size(); ++i) {
+            JsonObject account = minecraftAccounts.get(i).getAsJsonObject();
+            if (account.get("username").getAsString().equals(username)) {
+                return account;
+            }
+        }
+
+        return null;
+    }
+
+    public static void addMinecraftAccount(String username, String onlineUuid) {
+        String offlineUuid = PlayerEntity.getOfflinePlayerUuid(username).toString();
+        onlineUuid = formatUuid(onlineUuid);
+
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty("username", username);
+        jsonObject.addProperty("online-uuid", onlineUuid);
+        jsonObject.addProperty("offline-uuid", offlineUuid);
+        jsonObject.addProperty("timestamp", LocalDateTime.now().toString());
+
+        getMinecraftAccounts().add(jsonObject);
+        saveCache();
+    }
+
+    private static void removeExpired() {
+        JsonArray minecraftAccounts = getMinecraftAccounts();
+        Collection<JsonObject> invalidAccounts = Sets.newHashSet();
+        for (int i = 0; i < minecraftAccounts.size(); ++i) {
+            JsonObject account = minecraftAccounts.get(i).getAsJsonObject();
+            LocalDateTime parsedTimestamp;
+            try {
+                parsedTimestamp = LocalDateTime.parse(account.get("timestamp").getAsString());
+            } catch (Exception e) {
+                continue;
+            }
+            boolean valid = Duration.between(parsedTimestamp, LocalDateTime.now()).toDays() <= 14;
+            if (!valid) {
+                invalidAccounts.add(account);
```

Comment:
You can just remove it here?

Reply:
Wouldn't that cause a CME?

Reply:
> Wouldn't that cause a CME?

I'm sorry, what does CME stand for?

Reply:
ConcurrentModificationException. Usually happens when modifying something from an array that is being iterated over, but I'm not sure if it does happen with JsonArrays

Reply:
It may. My bad.